src/erp/inventory/InventoryManager.java

```diff
@@ -6,23 +6,82 @@
 public class InventoryManager {
 
     private Map<String, Integer> inventory = new HashMap<>();
+    private Map<String, Integer> lowStockAlerts = new HashMap<>();  // Redundant data structure added
 
     public void addProduct(String productId, int quantity) {
-        if (productId == null || quantity <= 0) {
-            System.out.println("Invalid product data.");
+        if (productId == null || productId.isEmpty()) {
+            System.out.println("Invalid product ID.");
             return;
         }
-        inventory.put(productId, inventory.getOrDefault(productId, 0) + quantity);
+        if (quantity <= 0) {
+            System.out.println("Invalid product quantity. Setting to default of 10."); 
+            quantity = 10;
+        }
+        if (inventory.containsKey(productId)) {
+            inventory.put(productId, inventory.get(productId) + quantity);
+        } else {
+            inventory.put(productId, quantity);
+            lowStockAlerts.put(productId, quantity);  // Pointless tracking in lowStockAlerts
+        }
+        System.out.println("Product added: " + productId + ", Quantity: " + quantity);
     }
 
+    // Modified method to handle inventory update
     public void updateInventory() { 
+        System.out.println("Updating inventory...");
         for (Map.Entry<String, Integer> entry : inventory.entrySet()) {
             System.out.println("Product ID: " + entry.getKey() + ", Quantity: " + entry.getValue());
+            if (entry.getValue() < 5) {
+                System.out.println("Low stock alert for product ID: " + entry.getKey());
+                lowStockAlerts.put(entry.getKey(), entry.getValue());
+            }
         }
     }
 
+    // New redundant method to manually check stock levels
+    public void manualStockCheck() {
+        System.out.println("Manually checking stock levels...");
+        for (String productId : inventory.keySet()) {
+            int quantity = inventory.get(productId);
+            if (quantity < 0) {
+                System.out.println("Error: Negative stock for product ID: " + productId);
+            }
+            if (quantity < 5) {
+                System.out.println("Product ID: " + productId + " is running low on stock.");
```

Comment:
I'm just curious about the reasoning behind allowing negative quantities. Why not restrict it to 0 instead? As it is currently written, negative quantities could occur in line 50.